### ISSUE TITLE
Move Documentation website to Docusaurus or Hugo

### ISSUE THREAD

**Issue by oleg-nenashev** [wiremock/wiremock.org-sources#24]
Currently the documentation uses Jekyll, and it is pretty weak when it comes to product-grade documentation (macros, external sourcing for docs-as-code, etc., etc.). I think it would be preferable to have another documentation engine, preferably Docusaurus or Hugo. I find Docusaurus much more powerful and sustainable, but IIUIC @tomakehurst prefers Hugo. Subject for a discussion

## References

- https://docusaurus.io/
- https://gohugo.io/

**Comment by oleg-nenashev** [comment]
TL;DR: This week in my head

![Battle of documentation tools](https://user-images.githubusercontent.com/3000480/233710719-bdf4fdd6-ca06-4bcd-b90b-142317f5684d.png)

**Comment by SimonVerhoeven** [comment]
Antora's quite pleasant to work with

**Comment by oleg-nenashev** [comment]
@SimonVerhoeven Yes, indeed. The biggest problem there is need to have Markdown, and it creates quite a lot of issues with Antora. I have a pending discussion about this on Antora Zulip ([here](https://antora.zulipchat.com/#narrow/stream/282400-users/topic/Antora.20site.20with.20Markdown-ish.20sources)). You can find my evaluation notes [here](https://docs.google.com/document/d/1TAXUFw_GCcGeSlXYpIqdtZJbuULPhGN7pI0QM1C_4Hk/edit?usp=sharing) too. If  I do not manage to sort Markdown support in a reasonable way, might end up with MkDocs + Multirepo
